### DISCUSSION TITLE
Unable to use private key with password for Raft secure configuration

### DISCUSSION THREAD

**Discussion by promiteyka** [ClickHouse/ClickHouse#69536] [Q&A]
I'm trying to set up encrypted connections for a raft cluster.
This is a part of my keeper_config.xml with settings about keys. I have password on private key and set it up as written in the official documentation for Clickhouse.
```xml
  <openSSL>
        <server>
            <!-- Used for secure tcp port -->
            <certificateFile>/etc/clickhouse-server/ssl/clickhouse2_chain.pem</certificateFile>
            <privateKeyFile>/etc/clickhouse-server/ssl/clickhouse2_.key</privateKeyFile>
            <verificationMode>none</verificationMode>
            <loadDefaultCAFile>true</loadDefaultCAFile>
            <privateKeyPassphraseHandler>
              <name>KeyFileHandler</name>
               <options>
                <password>changeme</password>
               </options>
            </privateKeyPassphraseHandler>
            <cacheSessions>true</cacheSessions>
            <disableProtocols>sslv2,sslv3</disableProtocols>
            <preferServerCiphers>true</preferServerCiphers>
        </server>
    </openSSL>
```
this block of config works great for the 
```xml
 <keeper_server>
     <tcp_port_secure>9281</tcp_port_secure>
```
but when i enable encryption for raft cluster with 
```xml
      <raft_configuration>
           <secure>1</secure>
```
service fails with error 

0. ? @ 0x1773d58e in /usr/bin/clickhouse
1. nuraft::asio_service_impl::asio_service_impl(nuraft::asio_service_options const&, std::shared_ptr<nuraft::logger>) @ 0x177384ec in /usr/bin/clickhouse
2. nuraft::asio_service::asio_service(nuraft::asio_service_options const&, std::shared_ptr<nuraft::logger>) @ 0x1773a03b in /usr/bin/clickhouse
3. DB::KeeperServer::launchRaftServer(Poco::Util::AbstractConfiguration const&, bool) @ 0x14ea7a8e in /usr/bin/clickhouse
4. DB::KeeperServer::startup(Poco::Util::AbstractConfiguration const&, bool) @ 0x14eaa591 in /usr/bin/clickhouse
5. DB::KeeperDispatcher::initialize(Poco::Util::AbstractConfiguration const&, bool, bool, std::shared_ptr<DB::Macros const> const&) @ 0x14e93914 in /usr/bin/clickhouse
6. DB::TinyContext::initializeKeeperDispatcher(bool) const @ 0x14f3c78b in /usr/bin/clickhouse
7. DB::Keeper::main(std::vector<String, std::allocator<String>> const&) @ 0xe3fc015 in /usr/bin/clickhouse
8. Poco::Util::Application::run() @ 0x1792ee22 in /usr/bin/clickhouse
9. DB::Keeper::run() @ 0xe3f873d in /usr/bin/clickhouse
10. Poco::Util::ServerApplication::run(int, char**) @ 0x1794267a in /usr/bin/clickhouse
11. mainEntryClickHouseKeeper(int, char**) @ 0xe3f7213 in /usr/bin/clickhouse
12. main @ 0x89e8b13 in /usr/bin/clickhouse
13. __libc_start_main @ 0x2409b in /usr/lib/x86_64-linux-gnu/libc-2.28.so
14. _start @ 0x803036e in /usr/bin/clickhouse
 (version 23.3.2.37 (official build))
2024.09.11 23:04:25.450596 [ 26221 ] {} <Error> Application: std::exception. Code: 1001, type: boost::system::system_error, e.what() = use_private_key_file: BAD_DECRYPT (Cipher functions, OPENSSL_internal) [asio.ssl:503316581], Stack trace (when copying this message, always include the lines below)

if i remove password from private key, service started correctly and works as expected

I take a look on the tests from clickhouse repo https://github.com/ClickHouse/ClickHouse/blob/master/tests/integration/test_keeper_internal_secure/configs/ssl_conf.xml
But there are now password on a private key.

How to set up this scheme?

**Comment by UnamedRus** [comment]
It was merged 1 week ago, you need to check next release 24.9 (it will be released on this week)

https://github.com/ClickHouse/ClickHouse/pull/69582
